### PR TITLE
Move DataMovementManager to nnf-sos and rename to NnfDataMovementManager

### PR DESCRIPTION
**Pull request by bdevcich**
This moves DMM to nnf-sos so that nnf-sos can use the DMM. This enables
nnf-sos to check if the DMM is ready before proceeding with
copy_in/copy_out directives.

* Moved type.go file and rbac roles to nnf-sos
* Removed `crd/` - there are no data movement types defined here anymore
* Added README.md to `api` to explain the types are in nnf-sos
* Update references to NnfDataMovementManager and also NnfDataMovement
* Re-vendored nnf-sos

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -29,6 +29,6 @@ This repository was bootstrapped using the operator-sdk
 ```
```
Comment (roehrich-hpe) [inline Readme.md:29]: I think you can delete this seciton from the readme.

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -23,67 +23,67 @@ spec:
...
+  mountPath: '/mnt/nnf'
   template:
     spec:
       serviceAccountName: nnf-dm-controller-manager # not sure why 'controller-manager' isn't patched to include the 'nnf-dm-' prefix
       tolerations:
-      - key: "cray.nnf.node"
-        operator: "Equal"
-        value: "true"
-        effect: "NoSchedule"
+        - key: 'cray.nnf.node'
```
Comment (roehrich-hpe) [inline config/manager/manager.yaml:65]: You have a pretty-printer tool enabled that is making these PRs much larger and more noisy than necessary.  And it is reformatting manifests that have been auto-generated by kubebuilder.

**Review by roehrich-hpe — APPROVED** [review]

File: config/manager/manager.yaml (hunk 2 above)
Comment (bdevcich) [reply config/manager/manager.yaml:65]: I will figure out how to turn this off in vscode for yaml files

**Review by NateThornton — APPROVED** [review]
Are you doing this just to erase my existence from the codebase?

**Comment by bdevcich** [comment]
> Are you doing this just to erase my existence from the codebase?

Haha, yes 😉 

But also, we need sos to be able to check when dmm is ready. I have an additional PR coming that ties the worker daemonset status into DMM's ready flag. So when things like global lustre are still being mounted into the workers, sos doesn't try to do data movement.

**Pull request merged**